src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -475,9 +475,15 @@ private void AppendLineNoPromptInjection(ITextBuffer buffer)
             }
 
             /// <summary>Implements <see cref="IInteractiveWindowOperations2.TypeChar"/>.</summary>
-            internal void TypeChar(char typedChar)
+            public void TypeChar(char typedChar)
             {
-                InsertText(typedChar.ToString());
+                using (var transaction = UndoHistory?.CreateTransaction(InteractiveWindowResources.TypeChar))
```

Comment:
It seems that `Delete's` are not merged in editor though, but yeah, I think I will need to rely on `EditorOperations.InsertText` for TypeChar in order to get them mergable.

The main reason behind implement those operations myself is to have a consistent behavior and better undo/redo when dealing with intertwined input and readonly buffers in interactive window among all the delete related operations  (i.e. delete, cut, paste, insert text, etc.). Some of them could be passed on to `EditorOperations` but I prefer consistency here :)